python_modules/libraries/dagster-looker/dagster_looker/lkml/asset_utils.py

```diff
@@ -1,11 +1,104 @@
 from pathlib import Path
-from typing import List, Sequence
+from typing import AbstractSet, Any, List, Mapping, Sequence, Tuple
 
 import lkml
 import yaml
 from dagster import AssetSpec
+from dagster._core.utils import toposort_flatten
+
+from dagster_looker.lkml.dagster_looker_lkml_translator import (
+    DagsterLookerLkmlTranslator,
+    LookMLStructureType,
+)
+
+
+def deep_merge_objs(onto_obj: Any, from_obj: Any) -> Any:
+    """Deep merge, which on dictionaries merges keys, and on lists appends elements.
+    If a value is present in both onto_obj and from_obj, the value from from_obj is taken.
+    For lists, entries are deduplicated by "name" key, resolving conflicts in favor of from_obj.
+    """
+    if isinstance(onto_obj, dict):
+        if from_obj is None or from_obj is ...:
+            return onto_obj
+        if not isinstance(from_obj, dict):
+            raise Exception("Cannot merge a dictionary with a non-dictionary object")
+
+        result = {
+            key: deep_merge_objs(onto_obj.get(key, ...), from_obj.get(key, ...))
+            for key in onto_obj.keys() | from_obj.keys()
+        }
+
+        return result
+
+    if isinstance(onto_obj, list):
+        if from_obj is None or from_obj is ...:
+            return onto_obj
+        if not isinstance(from_obj, list):
+            raise Exception("Cannot merge a list with a non-list object")
+
+        from_obj_names = [obj["name"] for obj in from_obj if "name" in obj]
+        result = [
+            obj
+            for obj in onto_obj
+            if not isinstance(obj, dict) or obj.get("name") not in from_obj_names
+        ] + from_obj
+
+        return result
+
+    return from_obj if from_obj is not ... else (onto_obj if onto_obj is not ... else None)
+
+
+def postprocess_loaded_structures(
+    structs: List[Tuple[Path, LookMLStructureType, Mapping[str, Any]]],
```

Comment:
Maybe make a `TypeAlias` for `Tuple[Path, LookMLStructureType, Mapping[str, Any]]`?